### PR TITLE
minimal feature set

### LINKED ISSUE #1: Minimal feature set
Extend `flora`, `flora_model`, `flora_parser` and `flora_dot_gen` and/or add further crates to support the following:
* read from stdin
  * `class A; class B; class C;`
* write to stdout
  * DOT that when piped through GraphViz generates one rectangle per class with the class name in it and solid lines connecting every pair of rectangles (fully connected graph)

### PR DESCRIPTION
**Pull request by jeysal**
closes #1

### CODE REVIEW

File: dot_gen/src/graph/graph_walk.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        Cow::Owned(
+            self.0
+                .keys()
+                .cloned()
+                .enumerate()
+                .flat_map(|(i, from)| {
+                    self.0
+                        .keys()
+                        .cloned()
+                        .skip(i + 1)
```
Comment (iyzana) [inline dot_gen/src/graph/graph_walk.rs:19]: First `skip` then `cloned`

**Review by iyzana — CHANGES_REQUESTED** [review]

**Review by iyzana — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -5,57 +5,30 @@
...
-  * might be sufficient for simple class diagrams, sooner or later switch to option 2 will become necessary
-
-## Future
-* [Asciidoctor Diagram](https://asciidoctor.org/docs/asciidoctor-diagram/) integration?
-* Language Service
-* Editor syntax themes (nice to have)
-* styling (CSS-like?)
+## Usage
+
+Flora currently always reads from stdin and writes to stdout.
```
Comment (hesch) [inline README.md:10]: Shouldn't this be named FloraUML instead of Flora in this document?

Comment (jeysal) [reply README.md:10]: I mean, I don't see a chance that people will use the full name including "UML" anyway, so we might as well make it official straight away. "Node.js" is even shorter and people still omit the ".js" all the time.
Are there any concrete ambiguities caused by calling it "Flora"?

Comment (hesch) [reply README.md:10]: Not really, I just thought that we might want to stick to the "official" name FloraUML in the documentation. But we can use Flora to establish the abbreviation. This option would also be better as the name of the binary.

**Review by hesch — APPROVED** [review]

File: dot_gen/src/graph/mod.rs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+mod graph_walk;
+mod labeller;
+
+use flora_model::Model;
+use std::collections::BTreeMap;
+
+pub struct Graph<'a>(BTreeMap<usize, Node<'a>>);
+
+type Node<'a> = &'a str;
+type Edge = (usize, usize);
```
Comment (thetric) [inline dot_gen/src/graph/mod.rs:10]: Does `usize` refer to the Node ID?

File: parser/src/class/parser.rs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn single_character_name() {
+        assert_eq!(
+            class_declaration(Input("class A;")),
+            Ok((Input(""), ClassDeclaration { name: "A" }))
+        )
```
Comment (thetric) [inline parser/src/class/parser.rs:31]: Why is the second argument `Ok((Input(""), ClassDeclaration { name: "A" }))` and not just `ClassDeclaration { name: "A" }))`?

File: dot_gen/src/graph/mod.rs (hunk 3 above)
Comment (jeysal) [reply dot_gen/src/graph/mod.rs:10]: Correct, the key in the map above.

File: parser/src/class/parser.rs (hunk 4 above)
Comment (jeysal) [reply parser/src/class/parser.rs:31]: `nom` parsers return a tuple of the remaining input data and the parsed AST.

**Pull request merged**